src/main/webapp/app/exercises/programming/shared/code-editor/service/code-editor-repository.service.ts

```diff
@@ -58,6 +61,10 @@ const handleErrorResponse = <T>(conflictService: CodeEditorConflictStateService)
             if (err.status === 409) {
                 conflictService.notifyConflictState(GitConflictState.CHECKOUT_CONFLICT);
             }
+            // If we receive a timeout error, return ok and sync later
+            if (err.status === 504) {
```

Comment:
How should we handle HTTP Error 502 (Bad Gateway) here? It will be returned if the load balancer/proxy cannot reach Artemis.

Reply:
I guess there are more error codes that we should handle, only 409 means the server reported a conflict.
We might consider in a follow up that we use an else statement here instead